cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -211,8 +215,10 @@ public void deleteProfile(ProfileId profileId)
     Transactionals.execute(transactional, context -> {
       ProfileDataset profileDataset = getProfileDataset(context);
       Profile profile = profileDataset.getProfile(profileId);
-      AppMetadataStore appMetadataStore = AppMetadataStore.create(cConf, context, datasetFramework);
-      deleteProfile(profileDataset, appMetadataStore, profileId, profile);
+      TransactionRunners.run(transactionRunner, structuredTableContext -> {
```

Comment:
did you nest these two transactions on purpose? OK if it works. In other places you had separated them and added a TODO.

Reply:
yes, when it's more trouble than it's worth to separate - I just keep them together

Reply:
ok makes sense